lib/Configuration.js

```diff
@@ -263,15 +221,39 @@ export default class Configuration {
         normalizePath(fromFile, this.workingDirectory),
         normalizePath(appliesFrom, this.workingDirectory)
       );
-    })[key];
+    });
+    return this.mergedValue(
+      applyingConfigs.map((config: Object): any => config[key]),
+      key,
+      fromFile
+    );
+  }
 
-    if (typeof value === 'function') {
-      return value({
-        pathToCurrentFile: this.pathToCurrentFile,
-        pathToImportedModule: fromFile,
-      });
+  mergedValue(values: Array<any>, key: string, fromFile: string): any {
```

Comment:
This function barely uses `this`. What do you think about moving it to the top level and passing in `pathToCurrentFile` instead? That would make it a pure function.